### PR TITLE
Fix warnings

### PR DESCRIPTION
**Pull request by dennisklein**

### CODE REVIEW

File: base/event/FairMCEventHeader.cxx [hunk 1]
```diff
@@ -19,7 +19,7 @@
 FairMCEventHeader::FairMCEventHeader()
     : TNamed("MCEvent", "MC")
     , fRunId(0)
-    , fEventId(-1)
+    , fEventId(NoEventID)
```
Comment (ChristianTackeGSI) [inline base/event/FairMCEventHeader.cxx:22]: If you touch this member variable anyhow, could you move this to a member initializer?

File: base/event/FairMCEventHeader.h [hunk 2]
```diff
@@ -32,17 +36,22 @@ class FairMCEventHeader : public TNamed
...
 
     /** Standard constructor with run identifier **/
     FairMCEventHeader(UInt_t runId);
 
     /** Destructor **/
     virtual ~FairMCEventHeader();
 
+    /// Sentinel value for the event id, meaning no event id stored
+    static constexpr EventID NoEventID = std::numeric_limits<EventID>::max();
+
```
Comment (ChristianTackeGSI) [inline base/event/FairMCEventHeader.h:49]: Could you move this to the `using EventID`? That way all the EventID related stuff is in one place?

File: base/event/FairMCEventHeader.h [hunk 3]
```diff
@@ -77,7 +86,7 @@ class FairMCEventHeader : public TNamed
 
   protected:
     UInt_t fRunId;      ///  Run identifier
-    UInt_t fEventId;    ///  Event identifier
+    EventID fEventId;   ///  Event identifier
```
Comment (ChristianTackeGSI) [inline base/event/FairMCEventHeader.h:89]:
```suggestion
    EventID fEventId{NoEventID};   ///  Event identifier
```

**Review by ChristianTackeGSI — COMMENTED** [review]
Just some ideas.

File: base/event/FairMCEventHeader.cxx (hunk 1 above)
Comment (dennisklein) [reply base/event/FairMCEventHeader.cxx]: that is a member initializer :P you mean default member intialization? done

File: base/event/FairMCEventHeader.cxx [hunk 4]
```diff
@@ -57,7 +55,6 @@ FairMCEventHeader::FairMCEventHeader(Int_t iEvent,
                                      Int_t nPrim)
     : TNamed("MCEvent", "MC")
     , fRunId(0)
-    , fEventId(iEvent)
```
Comment (ChristianTackeGSI) [inline base/event/FairMCEventHeader.cxx:60]: I doubt you wanted to remove that one really?

I wonder, why this didn't throw a warning about unused parameter though…

File: base/event/FairMCEventHeader.h [hunk 5]
```diff
@@ -76,18 +85,18 @@ class FairMCEventHeader : public TNamed
...
-    Double32_t fZ;      ///  Primary vertex z [cm]
-    Double32_t fT;      ///  Event time [s]
-    Double32_t fB;      ///  Impact parameter [fm] (if relevant)
-    Int_t fNPrim;       ///  Number of input tracks
-    Bool_t fIsSet;      ///  Flag whether variables are filled
-    Double32_t fRotX;   ///  Rotation around x-axis (beam tilt) [rad]
-    Double32_t fRotY;   ///  Rotation around y-axis (beam tilt) [rad]
-    Double32_t fRotZ;   ///  Rotation around z-axis (event plane) [rad]
+    UInt_t fRunId;                  ///  Run identifier
+    EventID fEventId = NoEventID;   ///  Event identifier
```
Comment (ChristianTackeGSI) [inline base/event/FairMCEventHeader.h:89]: We once had a small discussion with @fuhlig1 that our _default_ (yes, I meant those) member initializers should have a consistent style using brances.

```suggestion
    EventID fEventId{NoEventID};   ///  Event identifier
```

I wonder, whether there is a clang-tidy / format option to actually enforce this. Didn't find one quickly.

Comment (dennisklein) [reply base/event/FairMCEventHeader.h]: done

**Review by ChristianTackeGSI — APPROVED** [review]
No idea, if you want to backport to 18.8?

**Comment by dennisklein** [comment]
> No idea, if you want to backport to 18.8?

That was the plan, see the assigned milestone :P

**Review by karabowi — APPROVED** [review]

**Pull request merged**